hydromt/gis_utils.py

```diff
@@ -553,3 +558,46 @@ def to_geographic_bbox(bbox, source_crs):
         bbox = Transformer.from_crs(source_crs, target_crs).transform_bounds(*bbox)
 
     return bbox
+
+
+def prepare_pyogrio_reader_filters(
+    fn: IOBase,
+    bbox: GEOM_TYPES | None = None,
+    mask: GEOM_TYPES | None = None,
+    crs: CRS | None = None,
+) -> Tuple[float, float, float, float] | None:
+    """Create a pyogrio-compatible bbox filter.
+
+    Pyogrio does not accept a mask, and requires a bbox in the same CRS as the data.
+    This function takes the possible bbox filter, mask filter and crs of the input data
+    and returns a bbox in the same crs as the data based on the input filters.
+
+    Parameters
+    ----------
+    fn: IOBase,
+        opened file
+    bbox: GeoDataFrame | GeoSeries | BaseGeometry
+        bounding box to filter the data while reading
+    mask: GeoDataFrame | GeoSeries | BaseGeometry
+        mask to filter the data while reading
+    crs: pyproj.CRS
+        coordinate reference system of the bounding box or geometry. If already set,
+        this argument is ignored.
+    """
+    if bbox is not None and mask is not None:
+        raise ValueError(
+            "Both 'bbox' and 'mask' are provided. Please provide only one."
+        )
+    if bbox is None and mask is None:
+        return None
```

Comment:
Should we also raise a warning here?

Reply:
It can be fine to not filter (global models?)

Reply:
Good point.